src/transformers/generation_utils.py

```diff
@@ -927,43 +927,43 @@ def generate(
             max_new_tokens (`int`, *optional*, defaults to None):
                 The maximum numbers of tokens to generate, ignore the current number of tokens. Use either
                 `max_new_tokens` or `max_length` but not both, they serve the same purpose.
-            min_length (`int`, *optional*, defaults to 10):
+            min_length (`int`, *optional*, defaults to `model.config.min_length`):
```

Comment:
```suggestion
            min_length (`int`, *optional*, defaults to `model.config.min_length` or 10 if the config does not set any value):
```

Reply:
This is verbose... but possibly the best solution until it gets fixed.

As @JoaoLages, the warning is already there (PT only). However, because our documentation shows the parameters first and generate has 39 parameters (😅 ), the warning is not visible at first (see [the docs](https://huggingface.co/docs/transformers/v4.21.0/en/main_classes/text_generation#transformers.generation_utils.GenerationMixin.generate)).

Reply:
Yes, that's why I'm suggesting this compromise :-)

Reply:
I'll make the changes then. Thanks for your great work HuggingFacers! 💪